### DISCUSSION TITLE
Hello. Does the latency shown in this figure mean that POPET's prediction takes 6/18 clock cycles?

### DISCUSSION THREAD

**Discussion by szuhzw** [CMU-SAFARI/Hermes#38] [Q&A] [answered]
![image](https://github.com/user-attachments/assets/547320c6-a0e0-41d5-9470-2ee66e69bd90)

**Comment by szuhzw** [comment]
HERMES_O = --config=$(HERMES_HOME)/config/hermes_base.ini --ddrp_req_latency=6
HERMES_P = --config=$(HERMES_HOME)/config/hermes_base.ini --ddrp_req_latency=18

What does ddrp_req_latency refer to?   @rahulbera

**Comment by rahulbera** [comment, marked as answer]
It's not the latency of the prediction, but to issue the speculative request directly to the main memory controller from the core. In modern multi-core SoCs, even a request generated inside the core may take several cycles to travel to the main memory controller depending on the on-chip network/datapath design. Please refer to the end of Sec. 7.2 and Sec. 8.4.3 in the paper.
